File: 2021/src/test/java/de/jfriemel/aoc/days/Day07Test.java
package de.jfriemel.aoc.days;

import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.Assert.assertEquals;

public class Day07Test {

    private static final String INPUT_STRING = """
            16,1,2,0,4,2,7,1,2,14""";

    private static final List<String> INPUT = Arrays.asList(INPUT_STRING.split("\n"));

    @Test
    public void testPart1() {
        String result = new Day07().part1(INPUT);
        assertEquals("37", result);
    }

    @Test
    public void testPart2() {
        String result = new Day07().part2(INPUT);
        assertEquals("168", result);
    }

}

File: 2023/src/test/kotlin/de/jfriemel/aoc/days/Day07Test.kt
package de.jfriemel.aoc.days

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class Day07Test {
    private val input = """
        32T3K 765
        T55J5 684
        KK677 28
        KTJJT 220
        QQQJA 483
    """.trimIndent().split("\n")

    @Test
    fun testPart1() {
        val result = Day07.part1(input)
        Assertions.assertEquals("6440", result)
    }

    @Test
    fun testPart2() {
        val result = Day07.part2(input)
        Assertions.assertEquals("5905", result)
    }
}
